### ISSUE TITLE
Feature request 

### ISSUE THREAD

**Issue by russoandrewrusso** [bwp91/homebridge-ewelink-sonoff#6]
I have two feature requests that I think would help a lot of people. This first is on a 4CH sonoff relay to have the ability to hide some of the channels and not expose them to HomeKit. The second request would be the ability to use a sonoff 4ch and use one contact for open one for close and one for stop. This allows you to stop a door/shade/screen in the middle.

**Comment by thepotterfamily** [comment]
Hi
Thanks for your suggestions - they are always welcome!
1. I will look into hiding unused channels. There is no way of knowing from the data from ewelink whether a channel is actually being used, so it would require some extra entries in the config file. But certainly possible.
2. Again, possible. Would you have a device that we could test with if I were to implement this?

**Comment by thepotterfamily** [comment]
With regards to #1 hiding a single channel. Please try updating to 2.3.0 and then in your config add the final line of the following code:
```json
"countryCode": *,
"username": *,
"password": *,
"hideFromHB": "HERE",
```
I've included the countrycode, username and password lines just so you know where to put the hideFromHB line in relation.

And then replace HERE with ```100*******SW*```, replacing the asterisks with the ewelink device id and the final asterisk with the switch number (ie 1, 2, 3, 4) you want to hide, for example ```10009553c9SW1```.

If you want to hide multiple channels, just separate with a comma, eg, ```10009553c9SW1,10009553c9SW2```.

Restart Homebridge and they should be removed. It works for me but I'd be interested in your feedback on how it works.

**Comment by thepotterfamily** [comment]
Also note that single channel devices (SWX) and the master device (SW0) for multi channel devices can't be hidden

**Comment by russoandrewrusso** [comment]
Sweet the hiding switches worked great. Since I was only using one of the four switches I ended up hiding all of them and am using SW0 the “master” one to control the single switch from HomeKit.

**Comment by bwp91** [comment]
ah glad it's working well!
with regards to #2, do you have a device set up already that would work with 3 switches (up/down/stop)?
As I don't have a four channel device it would be difficult for me to implement this without your help.
If you're up for it then please contact send me an email.
i'll leave my email address in the next comment

**Comment by bwp91** [comment]
bwp91@icloud.com

**Comment by utester123123bb** [comment]
Hi, Your Plugin works like a charm. Thanks a lot! but is there any ability to change the type of accessory? I need a Garage Door opener for Sonoff switch but in homekit it's shown as switch or light or fan only.

**Comment by bwp91** [comment]
Hi @utester123123bb 
Yes you have the ability to change between switch/fan/light in Home app because behind the scenes they work the same way (basically either on or off).
Other accessories like garage doors/window blinds, let me explain, Home app gives you a "slider" between 0%-100% which makes it rather more complicated. 
Rather like trying to control an on/off light bulb with a slider from 0%-100%. It's just not user friendly.
The other thing with garage doors/window blinds is that there are so many different setups.
Person A has a garage door with ONE switch for up and down.
Person B has a device with TWO switches, one for up and one for down.
Person C had a device with THREE switches- up, down and stop midway.
An example of how it could go totally wrong: Person D puts the Home app slider to 10% to "close their door" and Home app sends a request to ewelink to close the door. Then they change it to 0% (whilst the door it now closed) but then Homekit will send another request to close the door. What happens to a garage door motor if it's asked to close a further 10% when in fact it's already closed.

In a nutshell, there is SO much that can go wrong.

Having said that, I am ongoingly thinking about how i can implement it as I know a lot of people would find it very useful. Totally understand that having a garage door appear as a garage door seems more natural than as a light switch.

@utester123123bb can I ask your setup with the garage door perhaps comparing to persons A B and C above?

Also, glad everything else is working like a charm! Thank you for mentioning that.

Ben

**Comment by utester123123bb** [comment]
Hi, I having a Hormann Door with an one button control. When I push it my garage door starts open, when I push it again it stops opening and freeze and when I push it again one more time it closing. If I push only one time my garage doors fully opens so I need push it again to close it.

**Comment by bwp91** [comment]
@utester123123bb 
Could probably implement this, which sonoff device do you currently use to control it?
I'm asking basically to find out:
- is it just a one channel device? (please say yes!)
- or is it a multi channel device? (if so please say there are no other devices connected to it!)
- or is it a multi channel device with other devices connected to different channels (WORST case scenario when it comes to coding - almost impossible)
Let me know!

**Comment by utester123123bb** [comment]
> @utester123123bb
> Could probably implement this, which sonoff device do you currently use to control it?
> I'm asking basically to find out:
> 
> * is it just a one channel device? (please say yes!)
> * or is it a multi channel device? (if so please say there are no other devices connected to it!)
> * or is it a multi channel device with other devices connected to different channels (WORST case scenario when it comes to coding - almost impossible)
>   Let me know!
I am using Sonoff DC 5v connected manually to my remote garage control (yep it's insane engineering :-)) so it will be cool  to see garage door opener in homekit with fake status opened or closed after some time, for example 10 seconds.

**Comment by bwp91** [comment]
@utester123123bb 
if you're happy to, please send me an email in my next comment so i can try sort this out

**Comment by bwp91** [comment]
bwp91@icloud.com

**Comment by bwp91** [comment]
Closing due to:
- inactivity from original user. @russoandrewrusso if you have any more issues/comments please do let me know
- garage door situation initially working for @utester123123bb
